src/Microsoft.DotNet.Darc/DarcLib/Helpers/DependencyFileManager.cs

```diff
@@ -567,52 +566,44 @@ public void UpdateVersionDetailsXmlSourceTag(XmlDocument versionDetails, SourceD
     /// <returns>Updated global.json file if was able to update, or the unchanged global.json if unable to</returns>
     private Dictionary<GitFileMetadataName, string> UpdateDotnetVersionGlobalJson(
         SemanticVersion incomingDotnetVersion,
-        JObject globalJson,
-        bool forceUpdate = false)
+        JObject targetGlobalJson)
     {
-        try
+        JToken pinnedToken = targetGlobalJson.SelectToken("tools.pinned");
+        if (pinnedToken != null && pinnedToken.Type == JTokenType.Boolean && pinnedToken.Value<bool>())
         {
-            // Check if tools.pinned is set to true, if so skip update
-            JToken pinnedToken = globalJson.SelectToken("tools.pinned");
-            if (pinnedToken != null && pinnedToken.Type == JTokenType.Boolean && pinnedToken.Value<bool>())
-            {
-                _logger.LogInformation("Skipping dotnet SDK update because tools.pinned is set to true in global.json");
-                return null;
-            }
+            _logger.LogInformation("Skipping dotnet SDK update because tools.pinned is set to true in the target." +
+                $"repo's {VersionFiles.GlobalJson}");
+            return null;
+        }
 
-            if (SemanticVersion.TryParse(globalJson.SelectToken("tools.dotnet")?.ToString(), out SemanticVersion repoDotnetVersion))
-            {
-                if (repoDotnetVersion.CompareTo(incomingDotnetVersion) < 0 || forceUpdate)
-                {
-                    Dictionary<GitFileMetadataName, string> metadata = [];
+        if (!SemanticVersion.TryParse(targetGlobalJson.SelectToken("tools.dotnet")?.ToString(), out SemanticVersion repoDotnetVersion))
+        {
+            _logger.LogError($"Could not find or parse token `tools.dotnet` in target repo's `{VersionFiles.GlobalJson}`." +
```

Comment:
not sure if this is an error too. We usually save `LogError` for exceptions, maybe a warning?

Reply:
Agreed, this might very well not be there and it's a valid case

Reply:
Actually, two cases - not found is OK, cannot parse should be an error